olegbetsun/booking_up_for_beauty.go

```diff
@@ -0,0 +1,52 @@
+package booking
+
+import (
+	"time"
+	"strconv"
+	"fmt"
+	)
+
+func ParseDate(layout string, date string) time.Time{
+	t,err := time.Parse(layout, date)
+	if err != nil {
+		fmt.Printf("%v \n", err)
+	}
+	return t
+}
+
+
+// Schedule returns a time.Time from a string containing a date
+func Schedule(date string) time.Time {
+	layout := "1/02/2006 15:04:05"
+	t:= ParseDate(layout, date)
+	return t
+}
+
+// HasPassed returns whether a date has passed
+func HasPassed(date string) bool {
+	layout := "January 2, 2006 15:04:05"
+	t := ParseDate(layout, date)
+	return t.Before(time.Now())
+}
+
+// IsAfternoonAppointment returns whether a time is in the afternoon
+func IsAfternoonAppointment(date string) bool {
+	layout := "Monday, January 2, 2006 15:04:05"
+	t:= ParseDate(layout, date)
+	noon := time.Date(t.Year(), t.Month(), t.Day(), 12, 0, 0, t.Nanosecond(), t.Location())
+	eve := time.Date(t.Year(), t.Month(), t.Day(), 17, 0, 0, t.Nanosecond(), t.Location())
+	return t.After(noon) && t.Before(eve)
```

Comment:
It's enough checking that the parsedDate `.Hour()` is between 12 and 18. You don't need to construct new dates.

Reply:
Agree, think about the amount of extra work is doing here the computer. It has to retrieve the different Year, Months, Day, Nanoseconds, Location, then build the `noon` and `eve` dates, then do whatever internal login it does at `After` and `Before`.

If you want to preserve some of the readability, you could assign `noonHour := 12` and `eveHour := 17` or something like that, so the conditional has words instead of magic numbers.

Reply:
time has an "Hour" method that returns an int, which would make the code a lot easier